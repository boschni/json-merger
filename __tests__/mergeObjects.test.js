const jsonMerger = require("../dist");
const { testConfig } = require("../__helpers__");

describe(".mergeObjects()", function () {
  test("should process the objects and return the result", function () {
    const object1 = {
      a: {
        $replace: "this should be the value of a",
      },
    };

    const object2 = {
      b: {
        $replace: "this should be the value of b",
      },
    };

    const result = jsonMerger.mergeObjects([object1, object2], testConfig());

    expect(result).toMatchSnapshot();
  });

  test("should use the config object", function () {
    const object1 = {
      a: {
        $replace: "this should be the value of a and not be pretty printed",
      },
    };

    const object2 = {
      b: {
        $replace: "this should be the value of b and not be pretty printed",
      },
    };

    const result = jsonMerger.mergeObjects(
      [object1, object2],
      testConfig({
        stringify: false,
      })
    );

    expect(result).toMatchSnapshot();
  });
});
