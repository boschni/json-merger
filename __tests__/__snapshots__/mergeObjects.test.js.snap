// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.mergeObjects() should process the objects and return the result 1`] = `
"{
	\\"a\\": \\"this should be the value of a\\",
	\\"b\\": \\"this should be the value of b\\"
}"
`;

exports[`.mergeObjects() should use the config object 1`] = `
Object {
  "a": "this should be the value of a and not be pretty printed",
  "b": "this should be the value of b and not be pretty printed",
}
`;
